### PR TITLE
More static_assert to threadwise_transfer

### PR DESCRIPTION
**Pull request by j4yan**
Add static assertions on `SliceLengths` and `ScalarPerVector` to threadwise-copy.

### CODE REVIEW

File: include/ck/utility/sequence.hpp [hunk 1]
```diff
@@ -606,6 +606,12 @@ struct sequence_map_inverse
                                            SeqMap::Size()>::type;
 };
 
+template <index_t... Xs, index_t... Ys>
+__host__ __device__ constexpr bool operator==(Sequence<Xs...>, Sequence<Ys...>)
+{
+    return (true && ... && (Xs == Ys));
```
Comment (asroy) [inline include/ck/utility/sequence.hpp:612]: return (true && ... && (Xs == Ys));

or

return ((Xs == Ys)&&...);

?

Comment (j4yan) [reply include/ck/utility/sequence.hpp:612]: Not sure what if `Xs` and `Ys` are empty.

**Comment by j4yan** [comment]
@qianfengz I've added some static_assert to threadwise transfer.

File: include/ck/utility/sequence.hpp (hunk 1 above)
Comment (j4yan) [reply include/ck/utility/sequence.hpp:612]: Found this:
http://eel.is/c++draft/temp.variadic#9
which shows that if they are empty, the folding expression is true for `&&`.

Will make the change.

**Comment by asroy** [comment]
@qianfengz This PR enable some check, which capture some bugs in reduction. Please try your new reduction PR would pass the check in this PR

**Comment by qianfengz** [comment]
> @qianfengz This PR enable some check, which capture some bugs in reduction. Please try your new reduction PR would pass the check in this PR
Just merged its P.R.   Need re-synchronization if any update

**Review by asroy — APPROVED** [review]

**Comment by asroy** [comment]
This PR is included by #125

**Pull request closed without merging**